### ISSUE TITLE
code navigation broken on project with custom toolchain and `--incompatible_enable_cc_toolchain_resolution --action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1`

### ISSUE THREAD

**Issue by redsun82** [bazelbuild/intellij#3319]
We have a project where we configured a custom toolchain (using `clang` as executable on posix) skipping the auto-configured one by bazel. We are setting the [`--incompatible_enable_cc_toolchain_resolution`](https://github.com/bazelbuild/bazel/issues/7260) and  [`--action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1`](https://github.com/bazelbuild/bazel/blob/0ba4caa5fc4bfd2d7f9af2a1653ef5b4bd83d176/tools/cpp/cc_configure.bzl#L48). On this project the intellij bazel pluging prints the following warning when synced:
```
Warning:Unable to access compiler executable
  "<path to bazel cache>/execroot/__main__/external/local_config_cc/gcc".
Check if it is accessible from the cmdline.
```
and as a consequence, code navigation is more or less completely broken. We do have a workaround, which is to link symbolically clang to the location indicated by the warning, but that's not ideal.

I managed to reduce our case to a rather small project presenting the same warning (on Linux with gcc). You can find it here:

https://github.com/redsun82/foo

I noticed dropping `--action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1` removes the warning, but my guess is that in that case the plugin will then use the host auto-configured compiler instead of the custom one.

**Comment by jesseschalken** [comment]
I believe the Bazel CLion plugin's toolchain resolution is still based on `--crosstool_top`, `--compiler` and `--cpu` and is yet to be upgraded to platforms and toolchains, and you will have to set those to an appropriate value to allow CLion to find the compiler. You can also use a [platform mappings](https://bazel.build/concepts/platforms-intro#platform-mappings) file.

**Comment by redsun82** [comment]
Thanks for the pointer, I was unaware of the platform mappings migration feature, I will definitely try that!

**Comment by jesseschalken** [comment]
I tried to tell CLion about my compiler through platform mappings yesterday and couldn't get it to work. Not sure if I was doing it right or my plugin is out of date but you might need to add the command line options directly to your `.bazelrc`.

**Comment by redsun82** [comment]
yeah, I ended up with the same result, and the same solution :smile: I added a `cc_toolchain_suite` target (which is not required with the platform-driven toolchain resolution) and the `--crosstool_top` option in `.bazelrc`, with notes to remove them when CLion allows it.

**Comment by sgowroji** [comment]
Hello @redsun82, Are you still looking support on the above issue. Thanks!

**Comment by LittleHuba** [comment]
I'm facing the same issue and assume that this is the underlying problem:
https://github.com/bazelbuild/bazel/issues/12712

**Comment by raoarun** [comment]
@maxwellgumley-brt @redsun82 is this still an issue?

**Comment by charles-toller** [comment]
Just tried version 2023.04.18.0.1-api-version-231 and am experiencing the same issue.

**Comment by garethellis0** [comment]
The example provided by @redsun82 produces the expected error on indexing, without any additional modification. @tpasternak - have you tried to reproduce on a clean system? 

Plugin Ver: `2023.11.23.0.1-api-version-233`  

```
Warning:Unable to access compiler executable "/home/gellis/.cache/bazel/_bazel_gellis/16ecdb2c16e491184f86f97272d7cbbe/external/local_config_cc/gcc".
Check if it is accessible from the cmdline.
```

**Comment by ujohnny** [comment]
@garethellis0 we both tried and for some reason the issue does not reproduce on our side. The last my attempt was 2 weeks ago and I started to think that it might be some stale issue, but sounds like it's not. Could you please provide more details on the bazel version you use, your system etc. that might help us? We can also chat about this one on [bazelbuild.slack.com](bazelbuild.slack.com) in #intellij channel, and any sensitive data might be shared via https://uploads.jetbrains.com

**Comment by ujohnny** [comment]
@garethellis0 I'm not sure if I will be able to address this before the Christmas since I have vacation starting middle of the next week and then will be available only from 2nd week 2024. There is a little chance that we can sort us some legal questions with Google, so I might look into this, but I can't promise that. Hope for your understanding. 

btw I tried to reproduce this with bazel 6.4.1, so this might be the case that I wasn't able to get it failed.

**Comment by garethellis0** [comment]
I just got around to testing the original repro from @redsun82 with bazel 6.4.0, and it seems the issue is addressed, the repro runs fine! IMO we can close this ticket out.
